p2p/node.go

```diff
@@ -381,6 +392,13 @@ func (n *QriNode) disconnected(_ net.Network, conn net.Conn) {
 	log.Debugf("disconnected from peer: %s", conn.RemotePeer())
 	pi := n.Host().Peerstore().PeerInfo(conn.RemotePeer())
 	n.pub.Publish(context.Background(), event.ETP2PPeerDisconnected, pi)
+
+	if n.profileService.IsQriPeer(pi.ID) {
+		pro, err := n.Repo.Profiles().PeerProfile(pi.ID)
+		if err == nil {
+			n.pub.Publish(context.Background(), event.ETP2PPeerDisconnected, pro)
+		}
+	}
```

Comment:
We're trying to reduce / remove direct use of the `Repo` interface so we can one day remove repo entirely. To that end, the new profiles service should own access to peer profiles. It'd passed a `Repo` instance on construction, and can manage these `PeerProfile` calls. 

Let's refactor `IsQriPeer` to be a `PeerProfile` accessor method that returns nil when no profile exists for a given peer.ID:

```suggestion
	if pro := n.profiles.PeerProfile(pi.ID); pro != nil {
		n.pub.Publish(context.Background(), event.ETP2PPeerDisconnected, pro)
	}
```

Reply:
Hmmm, what about ConnectedPeerProfile? Because we are checking if this is a connected peer, not a peer that we have potentially seen before. Or is it enough to say in the comment of `PeerProfiles` "only returns profiles of peers that we have connected to this session"

Reply:
Actually, refactored to have `profiles.HandleQriPeerDisconnect` and doing all the work of emitting the events and looking for the profile into the profile service. Now the `ETP2PPeerDisconnected ` and `ETP2PPeerConnected` events both get emitted from the same service.